src/eval/execution.rs

```diff
@@ -80,10 +155,11 @@ fn nix_value_from_module(
 
     let nix_value = call_js_function(scope, &nix_value, &[eval_ctx.into()])?;
 
-    let to_strict_fn: v8::Local<v8::Function> = try_get_js_object_key(scope, &nixrt, "toStrict")?
-        .expect("Could not find the function `toStrict` in `nixrt`.")
-        .try_into()
-        .expect("`n.toStrict` is not a function.");
+    let to_strict_fn: v8::Local<v8::Function> =
+        try_get_js_object_key(scope, &nixrt, "recursiveStrict")?
```

Comment:
I wonder if `recursiveStrict` is the right thing to call here. Actually, we should probably not call `toStrict` either.

We should only force strict evaluation if the user really wants to see the entire value. For example, if the user requested only a specific attrpath like `foo.bar.moo`, then only those values should be strictly evaluated.

I would suggest we move the strict evaluation out of this function, much higher up. What do you think? Should we move this "to strict" bit of logic to the `evaluate` function?

Reply:
The `nix_value_from_module` function is only called for the final value that's about to be sent over to Rust anyway, for the root module (only in the `evaluate` function), not per-import. Imports are still lazy (I should make a test for that sometime).

I don't think this should be the responsibility of this code specifically. In the future, if we allow the evaluator to only evaluate a specific attrpath, then that should be transpiled into js imo rather than being manually done in rust.